### ISSUE TITLE
Play application plugin: Migrate continuous & run integration tests

### ISSUE THREAD

**Issue by bmuschko** [gradle/playframework#38]
- Gradle core integration testing provided functionality for working on a handle of a Gradle test process
- TestKit does not easily support those scenarios
- We'll have to decide how to approach the problem without spending too much time

**Comment by bmuschko** [comment]
Implementing these tests will be hard with the capabilities provided by TestKit at the moment.

For verifying a running Play application the following functionality is needed:

1. TestKit can start a build in a non-blocking way so that output from that build can be inspected to determine the running port, implement health checks ("is application running", "did startup fail" etc.). This might be optional but I am not fully sure - it would be handy to retrieve some sort of handle from the non-blocking build for further inspection (see `org.gradle.integtests.fixtures.executer.GradleHandle`).
3. A previously started build can be gracefully canceled. One option would be to expose the cancellation token available in the tooling API to the TestKit API. The token should also provide an option to identify when the build is fully finished. We _might_ also need a cancellation option to just kill the build in case something goes wrong.

For continuous builds, the Gradle core fixture `org.gradle.integtests.fixtures.AbstractContinuousIntegrationTest` seems to indicate.

1. Similar requirements as for verifying running builds.
2. We might need a standard input pipe for sending input to the running build.
3. Apparently, we need access to another Gradle service called [`PendingChangesManager`](https://github.com/gradle/gradle/blob/master/subprojects/platform-play/src/integTest/groovy/org/gradle/play/integtest/continuous/AbstractPlayReloadIntegrationTest.groovy#L80).

Bottom line: The existing fixtures make it look like it's going to be a lot of effort to migrate the tests even if you provide all the functionality above. It's almost overwhelming to look at all the functionality in the fixtures.

**Comment by pioterj** [comment]
@big-guy What are your thoughts on unblocking this in a cost-efficient way?

**Comment by big-guy** [comment]
To unblock this, I think we should try doing this:

Once the smoke test #58 is working, we should add a version of these classes to platform-play in gradle/gradle that uses the playframework plugin:
- `PlayContinuousBuildIntegrationTest` 
- `PlayReloadWaitingIntegrationTest`
- `PlayContinuousBuildReloadIntegrationTest`
- `PlayContinuousBuildReloadWaitingIntegrationTest`

These tests should try to stay as close to the originals and just modify the build script as necessary.  This will serve as sort of smoke tests for continuous build for playframework in gradle/gradle and reuse all of our internal continuous build test infrastructure. 

If we want to narrow some of the differences between gradle/gradle and playframework, we could consider reusing PlayApplicationRunner and related classes from gradle/gradle until we can properly migrate the tests.  But I don't think that's necessary for now and living with the duplication might make it easier to make changes to playframework. 

Our process will be to update the gradle/gradle project whenever there's a new playframework release and if there's a breakage caused by Gradle internal changes, we'll deal with that as they come up.

**Comment by bmuschko** [comment]
I started by added the equivalent of `PlayContinuousBuildIntegrationTest` to get started: https://github.com/gradle/gradle/pull/8962
